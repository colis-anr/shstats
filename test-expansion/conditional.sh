#!/bin/sh

x=1
if true; then
    y=2
    echo $x ${y}
else
    y=3
    echo  ${x} $y
fi
echo $x $y
