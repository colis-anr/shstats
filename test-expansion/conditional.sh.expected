[
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix",
                [
                  "CmdPrefix_AssignmentWord",
                  [ [ "Name", "x" ], [ "Word", "1", [] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_CompoundCommand",
              [
                "CompoundCommand_IfClause",
                [
                  "IfClause_If_CompoundList_Then_CompoundList_ElsePart_Fi",
                  [
                    "CompoundList_Term_Separator",
                    [
                      "Term_AndOr",
                      [
                        "AndOr_Pipeline",
                        [
                          "Pipeline_PipeSequence",
                          [
                            "PipeSequence_Command",
                            [
                              "Command_SimpleCommand",
                              [
                                "SimpleCommand_CmdName",
                                [ "CmdName_Word", [ "Word", "true", [] ] ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ],
                    [
                      "Separator_SeparatorOp_LineBreak",
                      [ "SeparatorOp_Semicolon" ],
                      [ "LineBreak_Empty" ]
                    ]
                  ],
                  [
                    "CompoundList_NewLineList_Term_Separator",
                    [ "NewLineList_NewLine" ],
                    [
                      "Term_Term_Separator_AndOr",
                      [
                        "Term_AndOr",
                        [
                          "AndOr_Pipeline",
                          [
                            "Pipeline_PipeSequence",
                            [
                              "PipeSequence_Command",
                              [
                                "Command_SimpleCommand",
                                [
                                  "SimpleCommand_CmdPrefix",
                                  [
                                    "CmdPrefix_AssignmentWord",
                                    [ [ "Name", "y" ], [ "Word", "2", [] ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ],
                      [ "Separator_NewLineList", [ "NewLineList_NewLine" ] ],
                      [
                        "AndOr_Pipeline",
                        [
                          "Pipeline_PipeSequence",
                          [
                            "PipeSequence_Command",
                            [
                              "Command_SimpleCommand",
                              [
                                "SimpleCommand_CmdName_CmdSuffix",
                                [
                                  "CmdName_Word",
                                  [
                                    "Word",
                                    "echo",
                                    [ [ "WordName", "echo" ] ]
                                  ]
                                ],
                                [
                                  "CmdSuffix_CmdSuffix_Word",
                                  [ "CmdSuffix_Word", [ "Word", "1", [] ] ],
                                  [ "Word", "2", [] ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ],
                    [ "Separator_NewLineList", [ "NewLineList_NewLine" ] ]
                  ],
                  [
                    "ElsePart_Else_CompoundList",
                    [
                      "CompoundList_NewLineList_Term_Separator",
                      [ "NewLineList_NewLine" ],
                      [
                        "Term_Term_Separator_AndOr",
                        [
                          "Term_AndOr",
                          [
                            "AndOr_Pipeline",
                            [
                              "Pipeline_PipeSequence",
                              [
                                "PipeSequence_Command",
                                [
                                  "Command_SimpleCommand",
                                  [
                                    "SimpleCommand_CmdPrefix",
                                    [
                                      "CmdPrefix_AssignmentWord",
                                      [
                                        [ "Name", "y" ],
                                        [ "Word", "3", [] ]
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ],
                        [
                          "Separator_NewLineList",
                          [ "NewLineList_NewLine" ]
                        ],
                        [
                          "AndOr_Pipeline",
                          [
                            "Pipeline_PipeSequence",
                            [
                              "PipeSequence_Command",
                              [
                                "Command_SimpleCommand",
                                [
                                  "SimpleCommand_CmdName_CmdSuffix",
                                  [
                                    "CmdName_Word",
                                    [
                                      "Word",
                                      "echo",
                                      [ [ "WordName", "echo" ] ]
                                    ]
                                  ],
                                  [
                                    "CmdSuffix_CmdSuffix_Word",
                                    [ "CmdSuffix_Word", [ "Word", "1", [] ] ],
                                    [ "Word", "3", [] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ],
                      [ "Separator_NewLineList", [ "NewLineList_NewLine" ] ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName_CmdSuffix",
                [
                  "CmdName_Word",
                  [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                ],
                [
                  "CmdSuffix_CmdSuffix_Word",
                  [ "CmdSuffix_Word", [ "Word", "1", [] ] ],
                  [ "Word", "$y", [ [ "WordLiteral", "$y" ] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ]
]